### PR TITLE
moved platform_util.hpp, puff.hpp and union_endpoint.hpp to aux

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: include/libtorrent/aux_/platform_util.hpp [hunk 1]
```diff
@@ -1,11 +1,13 @@
 #ifndef TORRENT_PLATFORM_UTIL_HPP
 #define TORRENT_PLATFORM_UTIL_HPP
 
+#include "libtorrent/config.hpp"
```
Comment (arvidn) [inline include/libtorrent/aux_/platform_util.hpp:4]: if all you need is the `TORRENT_EXTRA_EXPORT` it's sufficient to include `aux_/export.hpp`

Comment (aldenml) [reply include/libtorrent/aux_/platform_util.hpp:4]: right, now the long wait of appveyor :)

File: include/libtorrent/aux_/platform_util.hpp [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 
+#include "libtorrent/aux_/export.hpp"
+
 #include <cstdint>
 
-namespace libtorrent {
+namespace libtorrent::aux {
 
-	int max_open_files();
+	TORRENT_EXTRA_EXPORT int max_open_files();
```
Comment (arvidn) [inline include/libtorrent/aux_/platform_util.hpp:10]: what prompted this? technically it's only necessary once there's a test that needs access to the function.
I think it's a good change, I'm just curious if you experienced an error without it.

**Review by arvidn — APPROVED** [review]

File: include/libtorrent/aux_/platform_util.hpp (hunk 2 above)
Comment (aldenml) [reply include/libtorrent/aux_/platform_util.hpp:10]: no error, was just mechanical thinking on my part

**Pull request merged**